### DISCUSSION TITLE
Gatsby StaticImage does not respect TailwindCSS responsive modifiers in the production build.

### DISCUSSION THREAD

**Discussion by kawpii** [gatsbyjs/gatsby#33565] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Following code work as expected in dev mode but Gatsby StaticImage do not respect TailwindCSS responsive modifiers in the production build.

```
import { StaticImage } from "gatsby-plugin-image";
import * as React from "react";

const IndexPage = () => {
  return (
    <main>
      <title>Home Page</title>
      <h1>Should be visible on mobile</h1>
      <StaticImage
        src="../images/icon.png"
        alt="logo"
        className="md:hidden"
        width={100}
      />
      <h1>Should be visible on tablet</h1>
      <StaticImage
        src="../images/icon.png"
        alt="logo"
        width={100}
        className="hidden md:block"
      />
    </main>
  );
};

export default IndexPage;

```

### Reproduction Link

https://github.com/kaushalyap/gatsby-tailwind-bug

### Steps to Reproduce

1.Clone the repo & `yarn install`
2. gatsby develop and go to `localhost:8000`
3. Stop the dev server and `gatsby build && gatsby serve`
...

### Expected Result

Production build UI should be consistent with the dev build UI, which means production build should respond to tailwindcss responsive modifiers as in the dev build

### Actual Result

In production build, both images are shown regardless of screen size

### Environment

```shell
System:
    OS: Linux 5.14 Fedora Linux 35 (Workstation Edition)
    CPU: (8) x64 Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 14.17.2 - /tmp/fnm_multishells/49671_1634538470105/bin/node
    Yarn: 1.22.15 - /usr/bin/yarn
    npm: 7.20.3 - /tmp/fnm_multishells/49671_1634538470105/bin/npm
  Languages:
    Python: 3.10.0 - /usr/bin/python
  Browsers:
    Firefox: 93.0
  npmPackages:
    gatsby: ^3.14.2 => 3.14.3
    gatsby-plugin-image: ^1.14.1 => 1.14.1
    gatsby-plugin-postcss: ^4.14.0 => 4.14.0
    gatsby-plugin-sharp: ^3.14.1 => 3.14.1
    gatsby-source-filesystem: ^3.14.0 => 3.14.0
    gatsby-transformer-sharp: ^3.14.0 => 3.14.0
```

### Config Flags

Nothing

**Comment by LekoArts** [comment, marked as answer]
This happens due to CSS specificity. The style for `.gatsby-image-wrapper-constrained` is for performance reasons in a `<style>` tag in the `<head>` and thus has a higher specificity than the CSS from another file (Tailwind). So it overwrites Tailwind for the `display` property.

Use `!important` and it'll work.

<img width="1392" alt="Bildschirmfoto 2021-10-18 um 09 47 51" src="https://user-images.githubusercontent.com/16143594/137689623-ed8385ff-ec1a-43a3-91c1-eb4ab754ac14.png">

**Reply by kawpii** [reply]
@LekoArts  Thanks, While this works, [setting important in TailwindCSS](https://tailwindcss.com/docs/configuration#important) increase CSS size due to it's utility approach.(addition of important to every utility class, where in TailwindCSS there are lot of utility classes.) even through unused classes are purged.

Any other way to set important! or fix this ?

**Reply by wardpeet** [reply]
What about adding another wrapper div with the necessary CSS? I don't think there is any other way to fix this than bumping specificity like @LekoArts said.

**Reply by kawpii** [reply]
@wardpeet  I'll try it, My question is why use two classes? Why not put all CSS attributes in `gatsby-image-wrapper` class? This way it will allow us to pass classes through `classNames` prop? While by default layout prop set to constrained, why not add class to existing wrapper depending on the layout prop value?

**Reply by wardpeet** [reply]
It's difficult cause you get different css styling based on the the layout you choose.

**Reply by hisnameisjimmy** [reply]
I'm running into the same issue here: https://github.com/gatsbyjs/gatsby/issues/33960, which is not resolved by setting important on the Tailwind config because Tailwind doesn't bring important to the aspect-ratio plugin.

**Reply by hisnameisjimmy** [reply]
Update, if you add `mode: 'jit'` to your Tailwind config, you can add !important to specific classes, [as specified here](https://tailwindcss.com/docs/just-in-time-mode#built-in-important-modifier). This solved the issue for me, although I still think there should be differences between develop and build versions of the site that cause these surprise headaches :)
